test/models/shop_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+require 'helpers/shop_lifecycle_test_helper'
+
+class ShopTest < ActiveSupport::TestCase
+  include ShopLifecycleTestHelper
+
+  def setup
+    @shop_1 = shops(:regular_shop)
+    @shop_2 = shops(:second_shop)
+  end
+
+  test "#uninstall marks the shop as uninstalled" do
+    assert_uninstalls @shop_1.shopify_domain do
```

Comment:
In this case we should directly check that the database destroy was completed instead of using the helper asserts.  This way the tests describe what the observable behaviour of uninstall is.  

```
@shop_1.uninstall
assert_nil Shop.find_by(id: @shop_1.id)
```